### PR TITLE
attach and detach accessibility tree root element

### PR DESCRIPTION
**Pull request by meganrogge**
Explains why navigation mode was not working and gets the output reading back to the state it was in - only announcing when text is too long to read

[172007](https://github.com/microsoft/vscode/issues/172007)

### CODE REVIEW

File: src/browser/AccessibilityManager.ts [hunk 1]
```diff
@@ -57,6 +57,7 @@ export class AccessibilityManager extends Disposable {
   ) {
     super();
     this._accessibilityTreeRoot = document.createElement('div');
+    this._terminal.element?.appendChild(this._accessibilityTreeRoot);
```
Comment (Tyriar) [inline src/browser/AccessibilityManager.ts:60]: Can we switch this to what it was before?

https://github.com/xtermjs/xterm.js/blob/19c760a20470c26666efad770fe82496c4a9bf1e/src/browser/AccessibilityManager.ts#L87

If that position isn't good, at least to the same line?

File: src/browser/AccessibilityManager.ts [hunk 2]
```diff
@@ -121,6 +122,7 @@ export class AccessibilityManager extends Disposable {
     // media query that drives the ScreenDprMonitor isn't supported
     this.register(addDisposableDomListener(window, 'resize', () => this._refreshRowsDimensions()));
     this.register(toDisposable(() => {
+      this._terminal.element?.removeChild(this._accessibilityTreeRoot);
       this._accessiblityBuffer.remove();
       this._accessibilityTreeRoot.remove();
```
Comment (Tyriar) [inline src/browser/AccessibilityManager.ts:127]: This should be covered already by `this._accessibilityTreeRoot.remove();`

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/browser/AccessibilityManager.ts (hunk 1 above)
Comment (meganrogge) [reply src/browser/AccessibilityManager.ts:60]: good find, yes 👍🏼 looks like that happened in Dec

File: src/browser/AccessibilityManager.ts [hunk 3]
```diff
@@ -121,6 +122,7 @@ export class AccessibilityManager extends Disposable {
     // media query that drives the ScreenDprMonitor isn't supported
     this.register(addDisposableDomListener(window, 'resize', () => this._refreshRowsDimensions()));
     this.register(toDisposable(() => {
+      this._terminal.element?.removeChild(this._accessibilityTreeRoot);
```
Comment (meganrogge) [inline src/browser/AccessibilityManager.ts:125]:
```suggestion
```

File: src/browser/AccessibilityManager.ts (hunk 1 above)
Comment (meganrogge) [inline src/browser/AccessibilityManager.ts:60]:
```suggestion
    this._terminal.element.insertAdjacentElement('afterbegin', this._accessibilityTreeRoot);
```

Comment (Tyriar) [reply src/browser/AccessibilityManager.ts]: Looks like my prototype broke it: https://github.com/xtermjs/xterm.js/commit/41dc93a008dc7bc4044020cfa483210891770b50

**Comment by meganrogge** [comment]
so this was due to our accessibility buffer changes

https://github.com/xtermjs/xterm.js/commit/41dc93a008dc7bc4044020cfa483210891770b50#diff-7904a8cc1e295a78532b41608480fddf931f80c485c53b78f99ae8cd5759dedcL86

**Pull request merged**